src/Corax/Indexing/IndexWriter.TextualFieldInserter.cs

```diff
@@ -510,12 +545,29 @@ public static void PrefetchContainerPages(IndexWriter writer, ref ContextBoundNa
                     continue;
 
                 long containerId = EntryIdEncodings.GetContainerId(cur);
-                pagesToPrefetch.Add(containerId / Voron.Global.Constants.Storage.PageSize);
+                postingListPagesProcessingBuffer.Add(containerId / Voron.Global.Constants.Storage.PageSize);
             }
 
-            pagesToPrefetch.Count = Sorting.SortAndRemoveDuplicates(pagesToPrefetch.RawItems, pagesToPrefetch.Count);
+            postingListPagesProcessingBuffer.Count = Sorting.SortAndRemoveDuplicates(postingListPagesProcessingBuffer.RawItems, postingListPagesProcessingBuffer.Count);
+
+            writer._transaction.LowLevelTransaction.DataPager.MaybePrefetchMemory(postingListPagesProcessingBuffer.GetEnumerator());
```

Comment:
Did you have a chance to look into that deeper during testing and see what is the actual "prefetching ratio"? Under the covers we have
```
CommandQueue = new BlockingCollection<PrefetchRanges>(128);
```

on which we call `TryAdd()` (might return `false`). Even before calling it we have `if (ShouldPrefetchSegment())`.

I wonder what is the effectiveness of prefetching in this scenario? Maybe we are running into some pattern / behavior where it's not so effective as we believe.

Reply:
Can we track how many page faults we are running here?